app/assets/javascripts/app/components/comment_form/comment_form.js

```diff
@@ -14,22 +15,33 @@
         comment: "=comment",
         discussion: "=discussion"
       },
+      controller: 'CommentFormCtrl as ctrl',
       templateUrl: 'components/comment_form/templates/comment_form.html',
-      link: function($scope, element, attrs) {
+    };
+  }
 
-        var clearBody = function() {
-          $scope.comment.body = '';
-        };
+  function CommentFormCtrl(DS, $scope) {
+    var ctrl = this;
 
-        var createComment = function(comment) {
-          $scope.discussion
-            .addComment(comment)
-            .then(clearBody);
-          $scope.toggle();
-        };
+    ctrl.clearBody = clearBody;
+    ctrl.updateComment = updateComment;
+    ctrl.createComment = createComment;
 
-        $scope.create = createComment;
-      }
-    };
+    function clearBody() {
+      $scope.comment.body = '';
```

Comment:
You can assign isolate scope's properties directly to the controller by adding ```bindToController: true``` to the *DDO*.
This means you can stop using `$scope`.

```
function commentForm() {
    return {
      ...,
      bindToController: true  
    };
  }

function CommentFormCtrl(DS) {
    ...
    function clearBody() {
      ctrl.comment.body = '';
    }

    function createComment(comment) {
     ctrl.discussion
        .addComment(comment)
        .then(clearBody);
      ctrl.toggle();
    }

    function updateComment(comment) {
      var updated = DS.createInstance('discussionComment', comment, { useClass:true });
      updated.update(comment.body);
      ctrl.toggle();
    }
  }
```